test/upgrade/probe_test.go

```diff
@@ -57,17 +60,61 @@ func TestContinuousEventsPropagationWithProber(t *testing.T) {
 
 	// Use zap.SugarLogger instead of t.Logf because we want to see failures
 	// inline with other logs instead of buffered until the end.
-	log := createLogger()
-	probe := prober.RunEventProber(context.Background(), log, client, config)
+	Log = createLogger()
+	p := prober.NewProber(Log, client, config)
+	p.Deploy(context.Background())
 	ensure.NoError(ioutil.WriteFile(test.EventingFlags.ReadyFile, []byte(readyMessage), 0666))
-	defer prober.AssertEventProber(context.Background(), t, probe)
+	defer removeProbeWithEventsValidation(t, p, config.FailOnErrors)
 
-	log.Infof("Waiting for file: %v as a signal that "+
+	Log.Infof("Waiting for file: %v as a signal that "+
 		"upgrade/downgrade is over, at which point we will finish the test "+
 		"and check the prober.", test.EventingFlags.PipeFile)
 	_, _ = ioutil.ReadFile(test.EventingFlags.PipeFile)
 }
 
+func removeProbeWithEventsValidation(t *testing.T, p prober.Prober, failOnErrors bool) {
+	ctx := context.Background()
+	p.Finish(ctx)
+	assertEventProber(t, p, failOnErrors)
+	p.Remove()
+}
+
+// assertEventProber will send finish event and then verify if all events propagated well
```

Comment:
seems this is just verifying and the Finish is done above? Maybe it's just left from copy pasting pervious version?